src/centralized_telescope/wrapper.rs

```diff
@@ -0,0 +1,38 @@
+use super::algorithm;
+use super::init::make_setup;
+use super::params::Params;
+use super::proof::Proof;
+use super::setup::Setup;
+use crate::utils::types::Element;
+
+/// The main ALBA struct with prove and verify functions.
+#[derive(Debug, Clone, Copy)]
+pub struct Wrapper {
+    setup: Setup,
+}
+
+impl Wrapper {
+    /// Initialize ALBA with `Params`.
+    pub fn create(params: &Params) -> Self {
+        let setup = make_setup(params);
+        Self::create_with_setup(&setup)
+    }
+
+    /// Initialize ALBA with `Setup`. This function is unsafe to use and should
+    /// be avoided.
+    pub fn create_with_setup(setup: &Setup) -> Self {
```

Comment:
I would add a `_unsafe` at the end of the name

Reply:
Preference between `create_with_setup_unsafe` and `create_unsafe`?

Reply:
Either is fine

Reply:
done